### PR TITLE
Change requests/diagnostics system update

### PR DESCRIPTION
**Pull request by StephenHodgson**
## Overview

Continued from https://github.com/XRTK/XRTK-Core/pull/426

## TODO 

- [x] populate text prefabs with console data in an efficient way
- [x] Ensure pin/unpin button works

### CODE REVIEW

File: XRTK.SDK/Packages/com.xrtk.sdk/Features/Diagnostics/Scripts/MixedRealityConsoleDiagnosticsHandler.cs [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+            for (int i = 0; i < textContainers.Length; i++)
+            {
+                textContainers[i] = Instantiate(logTextPrefab, transform).GetComponent<TextMeshProUGUI>();
+            }
+        }
+
+        /// <inheritdoc />
+        public void OnLogReceived(ConsoleEventData eventData)
+        {
+            // TODO fancy modulo to assign the next text prefab and scroll to it's position.
```
Comment (StephenHodgson) [inline XRTK.SDK/Packages/com.xrtk.sdk/Features/Diagnostics/Scripts/MixedRealityConsoleDiagnosticsHandler.cs]: @FejZa need help here.

Comment (FejZa) [reply XRTK.SDK/Packages/com.xrtk.sdk/Features/Diagnostics/Scripts/MixedRealityConsoleDiagnosticsHandler.cs]: Can you describe what needs to happen here?

Comment (FejZa) [reply XRTK.SDK/Packages/com.xrtk.sdk/Features/Diagnostics/Scripts/MixedRealityConsoleDiagnosticsHandler.cs]: Are you creating a new TextMeshProUGUI component "per log"?

Comment (StephenHodgson) [reply XRTK.SDK/Packages/com.xrtk.sdk/Features/Diagnostics/Scripts/MixedRealityConsoleDiagnosticsHandler.cs]: I am not creating a new component per log. 

I already populated the array in awake, and now I just want to populate the next item, but wrapped around the max entry item constraint in our log pool.

**Comment by StephenHodgson** [comment]
@FejZa thanks for the feedback.

> The logging ScrollRect is configured to scroll horizontally and vertically and has both Scrollbars. Is that intended? I think wrapping the text horizontally and just scrolling vertically as it used to be makes more sense?

Yes this was intended. So that anyone who decided they wanted to make their own could use what's pre-configured there. The scroll rect should be smart enough to display what's needed. From my tests it's only displaying the vertical since that's how the layout is driving it.

image

> The pin/unpin UI Element used to be a Toggle, now it's a button. Not an issue but the previous toggle based implementation would change the pin/unpin Icon depending on its state, so the UI indicates whether the window is pinned or not. This functionality is now lost.

Yeah I didn't get to see what component was used previously as all the references were missing when I tried to take a look at it before. So I just used the button. Should be fixed/addressed now.

> As suggested by you probably replacing the RadialView with another solves makes sense.

>The diagnostics window appears quite far away by default and eventually to small? This needs some more tweaking.

I polished these two feedback items a bit more. Let me know what you think. You're also more than welcome to push any tweaks you'd like :)

**Comment by FejZa** [comment]
Just tested and looking good :). I wasn't aware the ScrollRect will hide the horizontal one due to the layout. Actually so far never had to implement a horizontal ScrollRect so I always removed and disabled it. Nice!

**Comment by FejZa** [comment]
I merged the Core CR into my feature branch. This one seems ready as well the validation build is failing though. I'll check that.

**Comment by FejZa** [comment]
Seems it's just some Unity license activation issue. Never reached the build step actually.

**Pull request merged**